lib/phoenix/test/conn_test.ex

```diff
@@ -108,9 +108,20 @@ defmodule Phoenix.ConnTest do
   @doc """
   Creates a connection to be used in upcoming requests.
   """
-  @spec conn() :: Conn.t
+  @spec build_conn() :: Conn.t
+  def build_conn() do
+    build_conn(:get, "/", nil)
+  end
+
+  @doc """
+  Deprecated version of build_conn/0. Use build_conn/0 instead
+  """
+  @spec build_conn() :: Conn.t
   def conn() do
-    conn(:get, "/", nil)
+    IO.puts :stderr, """
+    warning: using conn/0 to build a connection is deprecated. Use build_conn/0 instead.
+    """
```

Comment:
`IO.write` to avoid duplicated new lines and pls include the stacktrace as well.

```elixir
IO.write :stderr, """
warning: ...
#{Exception.format_stacktrace}
"""
```